python/sqlalchemy/src/yuga_demo.py

```diff
@@ -0,0 +1,283 @@
+
+        try:
+            products = session.query(Product).all()
+            product_json = {
+                'products': [
+                    product.to_json() for product in products
+                ]
+            }
+            session.commit()
+            return product_json
+        except Exception as e:     
+            print('*** Exception in get_users: %s' % e)
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+
+
+class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
+
+    def _set_headers(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'json')
+        self.end_headers()
+
+    def _return_error(self):
+        self.send_response(400)
+
+    def do_GET(self):
+        print('do_Get called with path: %s' % self.path)
+        dao = UserOrdersDAO()
+
+        if self.path == '/users':
+            self.handle_list_users(dao)
+
+        if self.path == '/products':
+            self.handle_list_products(dao)
+
+        if self.path == '/list-products':
+            self.handle_list_all_products(dao)
+
+    def do_POST(self):
+
+        print('>> do_POST')
+        content_length = int(self.headers['Content-Length'])
+        body = json.loads(self.rfile.read(content_length))
+
+        print('body: ' % body)
+
+        dao = UserOrdersDAO()
+
+        print('doPost called with url: %s and data: %s' % (self.path, body))
+
+        try:
+            if self.path == '/users':
+                self.handle_create_user(dao, body)
+
+            if self.path == '/orders':
+                self.handle_create_order(dao, body)
+
+            if self.path == '/list-orders':
+                self.handle_list_orders(dao, body)
+
+            if self.path == '/products':
+                self.handle_create_product(dao, body)
+
+            if self.path == '/list-products':
+                self.handle_list_all_products(dao)
+        except: 
+            self._return_error()
+
+    # =======================================================
+    ## Handlers below
+    # =======================================================
+
+    def handle_create_user(self, dao, body):
+        self.handle_write_output(
+            dao.add_object(User(first_name=body['firstName'], last_name=body['lastName'], email=body['email'])))
+
+    def handle_list_users(self, dao):
+        self.handle_write_output(dao.get_users())
+
+    def handle_list_all_products(self, dao):
+        self.handle_write_output(
+            dao.list_products()
+        )
+
+    def handle_create_product(self, dao, body):
+        self.handle_write_output(
+            dao.create_product(body)
+        )
+
+    def handle_list_orders(self, dao, body):
+        self.handle_write_output(
+            dao.list_orders_for_user(body['userId'])
+        )
+
+    def handle_list_products(self, dao, body):
+        self.handle_write_output(
+            dao.list_orders_for_user(body['userId'])
```

Comment:
This seems to be wrong? We are listing orders instead of products.

Reply:
You are correct.  This was a copy-paste problem.  I have fixed it